block/block.go

```diff
@@ -175,18 +175,18 @@ func (m *Manager) applyBlock(block *types.Block, commit *types.Commit, blockMeta
 
 	m.blockCache.Delete(block.Header.Height)
 
-	if switchRole {
-		// TODO: graceful role change (https://github.com/dymensionxyz/dymint/issues/1008)
-		m.logger.Info("Node changing to proposer role")
-		panic("sequencer is no longer the proposer")
-	}
-
 	// validate whether configuration params and rollapp consensus params keep in line, after rollapp params are updated from the responses received in the block execution
 	err = m.ValidateConfigWithRollappParams()
 	if err != nil {
 		return err
 	}
 
+	// Check if there was an Update for the proposer and if I am the new proposer.
+	// If so, restart so I can start as the proposer.
```

Comment:
```suggestion
	// Check if there was an Update for the proposer and if I am the new proposer.
	// If so, restart so I can start as the proposer.
	// For current proposer, we don't want to restart because we still need to send the last batch.
	// This will be done as part of the `rotate` function. 
```